File: octave/linalg/vander_test.m
% Plot the Vandermonde matrix condition number given an increasing
% matrix order

ordermax = 20;
vcond = zeros(ordermax,1);
axis = 1:ordermax;
for n = axis
    x = zeros(1,n);
    for i = 0:n
        x(i+1) = i/n;
    end
    A = vander(x);
    vcond(n) = cond(A);
end
semilogy(axis, vcond);
pause;

File: python/vander_test.py
# Plot the Vandermonde matrix condition number given an increasing
# matrix order

import numpy as np
import matplotlib.pyplot as plt

ordermax = 20
vcond = []
axis  = list(range(1, ordermax+1)) 
for n in axis:
    x = np.empty(n+1)
    for i in range(0, n+1):
        x[i] = float(i)/n
    V = np.vander(x);
    vcond.append(np.linalg.cond(V))
plt.ylabel('log(cond)')
plt.xlabel('order')
plt.semilogy(axis, vcond)
plt.grid(True)
plt.show()
